attribute.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+# Represents a classroom.
+class Classroom
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    students << student
+    student.classroom = self
+  end
+end
+
+# Represents a Student.
+class Student
+  attr_accessor :name, :classroom
+
+  def initialize(name)
+    @name = name
+    @classroom = nil
+  end
+end
+
+# Represents a Book.
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+end
+
+# Represents a Rental.
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
+end
+
+# Represents a Person.
+class Person
+  attr_accessor :name, :rentals
+
+  def initialize(name)
+    @name = name
+    @rentals = []
+  end
+
+  def rent_book(book, date)
+    rental = Rental.new(date, book, self)
+    rentals << rental
+    book.rentals << rental
+  end
+end
```

Comment:
- [ ] As mentioned previously you don't have to redefine already created classes. For the person class here you should use the class in the `person.rb` file to setup the associations of a person and rentals.
- [ ] The `rent_book` method should only create an instance of a `Rental` so I'd advise removing the codes at line `60` and `61` as those associations will be  created in the `Rental` class (checkout the next comment 👇).